Sources/Exporters/OpenTelemetryProtocolHttp/logs/OtlpHttpLogExporter.swift

```diff
@@ -23,22 +24,28 @@ public class OtlpHttpLogExporter : OtlpHttpExporterBase, LogRecordExporter {
   }
   
   public func export(logRecords: [OpenTelemetrySdk.ReadableLogRecord], explicitTimeout: TimeInterval? = nil) -> OpenTelemetrySdk.ExportResult {
-    pendingLogRecords.append(contentsOf: logRecords)
-    let sendingLogRecords = pendingLogRecords
-    pendingLogRecords = []
-    
+  var sendingLogRecords: [ReadableLogRecord]!
+    dispatchQueue.sync {
+      pendingLogRecords.append(contentsOf: logRecords)
+      sendingLogRecords = pendingLogRecords
+      pendingLogRecords = []
+    }
+
     let body = Opentelemetry_Proto_Collector_Logs_V1_ExportLogsServiceRequest.with { request in
       request.resourceLogs = LogRecordAdapter.toProtoResourceRecordLog(logRecordList: sendingLogRecords)
     }
     
     var request = createRequest(body: body, endpoint: endpoint)
     request.timeoutInterval = min(explicitTimeout ?? TimeInterval.greatestFiniteMagnitude , config.timeout)
     httpClient.send(request: request) { [weak self] result in
+      guard let self = self else { return }
```

Comment:
this is wrong, we would ideally like to know if there was an error, which now is not printing anymore